pkgs/development/interpreters/shen-sbcl/default.nix

```diff
@@ -1,30 +1,39 @@
-{ lib, stdenv
+{ lib
+, stdenv
 , fetchurl
 , shen-sources
 , sbcl
 }:
 
 stdenv.mkDerivation rec {
-  name = "shen-sbcl-${version}";
+  pname = "shen-sbcl";
   version = "3.0.3";
+
   src = fetchurl {
     url = "https://github.com/Shen-Language/shen-cl/releases/download/v${version}/shen-cl-v${version}-sources.tar.gz";
     sha256 = "0mc10jlrxqi337m6ngwbr547zi4qgk69g1flz5dsddjy5x41j0yz";
   };
-  buildInputs = [
-    sbcl shen-sources
-  ];
-  buildPhase = ''
+
+  nativeBuildInputs = [ sbcl ];
+
+  preBuild = ''
     ln -s ${shen-sources} kernel
-    make sbcl
   '';
+
+  buildFlags = [ "build-sbcl" ];
+
+  checkTarget = "test-sbcl";
+
+  doCheck = true;
```

Comment:
Isn't `checkTarget = "test-sbcl";` necessary here?

Reply:
yes, it's necessary because `stdenv/setup` checkPhase only works with [`make check` and `make test`](https://github.com/marsam/nixpkgs/blob/c733722f9310744c89249d2e44b0cb170704d2dc/pkgs/stdenv/generic/setup.sh#L1060-L1067), and `shen-cl` makefile doesn't use those targets

Reply:
Ah yeah that was what I was meaning to point out, but I overlooked that you already had the correct `checkTarget`, nevermind.